### DISCUSSION TITLE
onSearch event is required

### DISCUSSION THREAD

**Discussion by OscarJVD** [JedWatson/react-select#5243] [Ideas]
onSearch event is necessary

**Comment by Rall3n** [comment]
Hello @OscarJVD,

could you elaborate on why such an event/prop is necessary?

**Reply by OscarJVD** [reply]
Because if it doesn't be there I can't show or alter State when I search

**Reply by OscarJVD** [reply]
It's to handle the search event and improve the functionality

**Reply by Rall3n** [reply]
As "searching" is a direct result of input change, can the same be achieved using the `onInputChange` prop?
